### PR TITLE
Forward port humble->iron

### PR DESCRIPTION
**Pull request by azeey**
# ➡️ Forward port

Port humble to iron
Not sure if we do forward ports in this repo, but I thought it was the best way to get the changes from #461, https://github.com/gazebosim/ros_gz/pull/458, and potentially other changes into `iron`.

Question:
* There were changes in `Changelog` files that I removed. Should I include them? Same for version changes in `package.xml` where the merge brought in 0.247. 

Branch comparison: https://github.com/gazebosim/ros_gz/compare/iron...humble

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

**Comment by mjcarroll** [comment]
> Not sure if we do forward ports in this repo

We do a little of both in this one because gazebo does forward ports and ros does backports.  It can be a bit muddled, but it works.

File: ros_gz/package.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <!-- TODO: Make this a metapackage, see
        https://github.com/ros2/ros2/issues/408 -->
   <name>ros_gz</name>
-  <version>0.247.0</version>
+  <version>0.245.0</version>
```
Comment (mjcarroll) [inline ros_gz/package.xml]: I'm not actually sure what we want to do here.  `245` is the latest release in iron and `247` is the latest in humble.

I think we want to keep the newer version and release it.  @ahcorde what do you think

Comment (ahcorde) [reply ros_gz/package.xml]: I think we should modify the version to `1.X`, `2.X`, ... sometimes is hard to find a version which is not already released

Comment (azeey) [reply ros_gz/package.xml]: According to https://github.com/gazebosim/ros_gz/blob/ros2/RELEASING.md#versioning, the versions for Iron should start with `0.254`. Do we want to keep that convension?

Comment (azeey) [reply ros_gz/package.xml]: I've reverted the change.

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**